### PR TITLE
Docstring convention enforcement marathon - part 2

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/mesh/meshing.py [hunk 1]
```diff
@@ -668,7 +676,11 @@ def create_gmsh_curve(buffer):
 
     @staticmethod
     def _fragment(
-        dim=[2, 1, 0], all_ent=None, tools=[], remove_object=True, remove_tool=True
+        dim: List[int] = [2, 1, 0],
+        all_ent=None,
```
Comment (oliverfunk) [inline bluemira/mesh/meshing.py:680]:
```suggestion
        all_ent: Optional[List[int]] = None,
```

Comment (oliverfunk) [inline bluemira/mesh/meshing.py:680]:
```suggestion
        all_ent: Optional[List[int]] = None,
```

File: bluemira/mesh/meshing.py [hunk 2]
```diff
@@ -690,7 +702,7 @@ def _fragment(
         return all_ent, oo, oov
 
     @staticmethod
-    def _map_mesh_dict(mesh_dict, all_ent, oov):
+    def _map_mesh_dict(mesh_dict: dict, all_ent, oov):
```
Comment (oliverfunk) [inline bluemira/mesh/meshing.py:705]:
```suggestion
    def _map_mesh_dict(mesh_dict: dict, all_ent, oov: list = []):
```

**Review by oliverfunk — APPROVED** [review]
LGTM, two small additional types in the comments

**Comment by CoronelBuendia** [comment]
> LGTM, two small additional types in the comments.
> 
> I've just noticed some inconsistent use of `dict` vs `Dict` (for example). How do you decide on when to use which?
> 
> We might need a larger discussion around moving to the lowercase types, after moving to py3.10+ As the `typing` module will be deprecated at some point https://docs.python.org/3/library/typing.html#deprecation-timeline-of-major-features

Bleurgh.. didnt know that.. We've been using upper case from typing for now, down the line I guess we'll have to change, its ok to have minor inconsistency. We discussed having a wider issue/discussion on typing.

**Pull request merged**